substratevm/src/com.oracle.svm.reflect/src/com/oracle/svm/reflect/serialize/SerializationSupport.java

```diff
@@ -108,9 +111,35 @@ public int hashCode() {
 
     private final Map<SerializationLookupKey, Object> constructorAccessors;
 
+    @Platforms(Platform.HOSTED_ONLY.class) private final Map<Class<?>, Boolean> deniedClasses;
```

Comment:
Adding any field to `SerializationSupport` that is not needed at image runtime is a design flaw. Please move these fields to `SerializationBuilder` where they actually belong.